client/consensus/manual-seal/src/tests.rs

```diff
@@ -0,0 +1,429 @@
+	));
+	let env = ProposerFactory::new(
+		client.clone(),
+		pool.clone(),
+		None,
+	);
+
+	// Spawn the background authorship task
+	let (mut sink, commands_stream) = futures::channel::mpsc::channel(1024);
+	let future = run_manual_seal(ManualSealParams {
+		block_import: client.clone(),
+		env,
+		client: client.clone(),
+		pool: pool.pool().clone(),
+		commands_stream,
+		select_chain,
+		consensus_data_provider: None,
+		inherent_data_providers,
+	});
+
+	thread::spawn(|| {
+		let mut rt = tokio::runtime::Runtime::new().unwrap();
+		rt.block_on(future);
+	});
+
+	// Submit a transaction to pool and verify the tx is processed okay
+	let result = pool.submit_one(&BlockId::Number(0), SOURCE, uxt(Alice, 0)).await;
+	assert!(result.is_ok());
+
+	// Send an engine command and ensure a block is generated
+	let (tx, rx) = futures::channel::oneshot::channel();
+	sink.send(EngineCommand::SealNewBlock {
+		parent_hash: None,
+		sender: Some(tx),
+		create_empty: false,
+		finalize: false,
+	}).await.unwrap();
+	let created_block = rx.await.unwrap().unwrap();
+	pool_api.increment_nonce(Alice.into());
+
+	assert_eq!(
+		created_block,
+		CreatedBlock {
+			hash: created_block.hash.clone(),
+			aux: ImportedAux {
+				header_only: false,
+				clear_justification_requests: false,
+				needs_justification: false,
+				bad_justification: false,
+				needs_finality_proof: false,
+				is_new_best: true
+			}
+		}
+	);
+
+	// ---
+	// Get the block
+	let block = client.block(&BlockId::Number(1)).unwrap().unwrap().block;
+	pool_api.add_block(block, true);
+
+	// Submit another tx
+	assert!(pool.submit_one(&BlockId::Number(1), SOURCE, uxt(Alice, 1)).await.is_ok());
+
+	let header = client.header(&BlockId::Number(1)).expect("db error").expect("imported above");
+	pool.maintain(sp_transaction_pool::ChainEvent::NewBestBlock {
+		hash: header.hash(),
+		tree_route: None,
+	}).await;
+
+	// Send another engine cmd
+	let (tx1, rx1) = futures::channel::oneshot::channel();
+	assert!(sink.send(EngineCommand::SealNewBlock {
+		parent_hash: Some(created_block.hash),
+		sender: Some(tx1),
+		create_empty: false,
+		finalize: false,
+	}).await.is_ok());
+
+	assert_matches::assert_matches!(rx1.await.expect("should be no error receiving"), Ok(_));
+
+	// ---
+	// Get the block
+	let block = client.block(&BlockId::Number(2)).unwrap().unwrap().block;
+	pool_api.add_block(block, true);
+	pool_api.increment_nonce(Alice.into());
+
+	// Submit another tx
+	assert!(pool.submit_one(&BlockId::Number(1), SOURCE, uxt(Alice, 2)).await.is_ok());
+
+	let (tx2, _rx2) = futures::channel::oneshot::channel();
+	assert!(sink.send(EngineCommand::SealNewBlock {
+		parent_hash: Some(created_block.hash),
+		sender: Some(tx2),
+		create_empty: false,
+		finalize: false,
+	}).await.is_ok());
+
+	// Ensure the forked block is in the db
+	// note: The following line fails the test
+	// let imported = rx2.await.unwrap().unwrap();
```

Comment:
@seunlanlege This is the line that fail when I run the test. Does it pass for you?

Reply:
When I was working on #7070 I noticed that one of the already-existing tests was failing. Is that the case here Jimmy? Did this test fail before you made changes?

Reply:
This is the test that fail 4/5 times approximately. Sometimes it pass...